akd/src/append_only_zks.rs

```diff
@@ -41,6 +47,26 @@ async fn tic_toc<T>(f: impl core::future::Future<Output = T>) -> (T, Option<f64>
     (f.await, None)
 }
 
+fn get_parallel_levels() -> Option<u8> {
+    #[cfg(not(feature = "parallel_insert"))]
+    return None;
+
+    #[cfg(feature = "parallel_insert")]
+    {
+        // Based on profiling results, the best performance is achieved when the
+        // number of spawned tasks is equal to the number of available cores.
+        // We therefore get the number of available threads and calculate the
+        // number of levels that should be executed in parallel to give the
+        // number of tasks closest to the number of threads. While there might
+        // be other tasks that are running on the threads, this is a reasonable
+        // approximation that should yield good performance in most cases.
+        let available_parallelism = std::thread::available_parallelism()
```

Comment:
is there a risk of getting this from std and not through some tokio-specific call? It seems strange to use a default parallelism crate and not tokio.

It seems like there might be a tokio specific way to do this?
https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html

Reply:
We are getting from std here, as it turns out we need tokio_unstable to use the linked API :/ There's no tokio-specific alternative AFAIK.

Reply:
i.e. `std::thread::available_parallelism()` is an approximation of the number of CPU cores, which is an approximation of the number of tokio threads. According to the [documentation](https://doc.rust-lang.org/std/thread/fn.available_parallelism.html) available parallelism takes into account CPU cores that have been reserved by the VM/host, which sounds perfect for our use-case.